samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.processor;
+
+import org.apache.samza.annotation.InterfaceStability;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.JobCoordinatorConfig;
+import org.apache.samza.config.MapConfig;
+import org.apache.samza.config.TaskConfigJava;
+import org.apache.samza.coordinator.JobCoordinator;
+import org.apache.samza.coordinator.JobCoordinatorFactory;
+import org.apache.samza.metrics.MetricsReporter;
+import org.apache.samza.task.AsyncStreamTaskFactory;
+import org.apache.samza.task.StreamTaskFactory;
+import org.apache.samza.util.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * StreamProcessor can be embedded in any application or executed in a distributed environment (aka cluster) as
+ * independent processes
+ * <p>
+ * <b>Usage Example:</b>
+ * <pre>
+ * StreamProcessor processor = new StreamProcessor(1, config);
+ * processor.start();
+ * try {
+ *  boolean status = processor.awaitStart(TIMEOUT_MS);    // Optional - blocking call
+ *  if (!status) {
+ *    // Timed out
+ *  }
+ *  ...
+ * } catch (InterruptedException ie) {
+ *   ...
+ * } finally {
+ *   processor.stop();
+ * }
+ * </pre>
+ */
+@InterfaceStability.Evolving
+public class StreamProcessor {
```

Comment:
Can one JVM process creates multiple StreamProcessors? Can the StreamProcessors being created in multiple threads? Is it thread safe?